buildbot/buildbot/worker.py

```diff
@@ -285,17 +290,27 @@ def stage_seashell(db, config):
             f.write(hls_code)
 
 
-def _sds_cmd(prefix, func_hw, c_hw, platform):
+def _sds_cmd(prefix, func_hw, c_hw, task):
     """Make a sds++ command with all our standard arguments.
     """
-    return prefix + [
-        'sds++',
-        '-sds-pf', platform,
-        '-sds-hw', func_hw, c_hw, '-sds-end',
-        '-clkid', '3',
-        '-poll-mode', '1',
-        '-verbose', '-Wall', '-O3',
-    ]
+    if not task['directives']:
+        return prefix + [
+            'sds++',
+            '-sds-pf', task['platform'],
+            '-sds-hw', func_hw, c_hw, '-sds-end',
+            '-clkid', '3',
+            '-poll-mode', '1',
+            '-verbose', '-Wall', '-O3',
+        ]
+    else:
+        return prefix + [
+            'sds++',
+            '-sds-pf', task['platform'],
+            '-sds-hw', func_hw, c_hw, '-hls-tcl', task['directives'], '-sds-end',
+            '-clkid', '3',
+            '-poll-mode', '1',
+            '-verbose', '-Wall', '-O3',
+        ]
```

Comment:
Let's avoid duplication here by building the constant part of the list and adding in only the necessary parts when there are directives.